starknet_py/net/l1/contracts.py

```diff
@@ -0,0 +1,24 @@
+from web3 import Web3
+from web3.contract import Contract as Web3Contract
+from web3.providers import BaseProvider as L1Provider
+
+from starknet_py.net.l1.networks import L1Network
+from starknet_py.net.l1.starknet_core_abi import STARKNET_CORE_L1_ABI
+
+
+class StarknetL1Contract:
+    @staticmethod
+    def on_l1_net(l1_net: L1Network, l1_net_provider: L1Provider) -> Web3Contract:
```

Comment:
We already have a chain parameter (for instance when creating a Contract), can we reuse it here?